Code/GraphMol/Canon.cpp

```diff
@@ -1358,5 +2154,144 @@ void canonicalizeEnhancedStereo(ROMol &mol,
+      chiralAtoms.push_back(atom);
+    }
+  }
+  for (auto &bond : mol.bonds()) {
+    if (bond->getStereo() == Bond::BondStereo::STEREOATROPCW ||
+        bond->getStereo() == Bond::BondStereo::STEREOATROPCCW) {
+      chiralBonds.push_back(bond);
+    }
+  }
+
+  if (!chiralAtoms.empty() || !chiralBonds.empty()) {
+    sgs.emplace_back(StereoGroupType::STEREO_ABSOLUTE, chiralAtoms,
+                     chiralBonds);
+  }
+  mol.setStereoGroups(sgs);  // could be empty, or have one abs group
+}
+
+void clearStereoGroups(ROMol &mol) {
+  // all chiral centers are added to an abs group
+  // if there are not chiral centers, no group is added
+  std::vector<StereoGroup> sgs;
+  mol.setStereoGroups(sgs);
+}
+
+void canonicalizeStereoGroups(std::unique_ptr<RWMol> &mol,
+                              StereoGroupAbsOptions outputAbsoluteGroups) {
+  // this returns a mol that has a caononical rep for the enhanced stereo
+  // groups it expands the given mol to all possible non-stereo-group mols,
+  // then determines a single set of stereo groups that uniquely represent
+  // that group.
+
+  // if there are both OR and AND groups, the AND groups are done first
+  // by haveing the working routine call itself for pre-prosing the AND
+  // groups
+
+  auto sgCount = mol->getStereoGroups().size();
+  if (sgCount == 0 ||
+      (sgCount == 1 && mol->getStereoGroups()[0].getGroupType() ==
+                           StereoGroupType::STEREO_ABSOLUTE)) {
+    if (outputAbsoluteGroups == StereoGroupAbsOptions::AlwaysInclude) {
+      addSingleAbsGroup(*mol);
+    } else {
+      clearStereoGroups(*mol);
+    }
+    return;
+  }
+
+  //  see if it is a simple compound, which has only one stereo group (not
+  //  abs)
+  // and if has two or fewer atoms in the group, and all chiral atoms are in
+  // that group.
+  //
+  // further more, if the findMeso routine matches the simple groups, it
+  // should be removed
+  //
+
+  if (sgCount == 1) {
+    auto &sg = mol->getStereoGroups()[0];
+    if (sg.getAtoms().size() <= 2 && sg.getBonds().size() == 0) {
+      bool isSimple = true;
+
+      for (auto &atom : mol->atoms()) {
+        if ((atom->getChiralTag() == Atom::ChiralType::CHI_TETRAHEDRAL_CCW ||
+             atom->getChiralTag() == Atom::ChiralType::CHI_TETRAHEDRAL_CW) &&
+            !boost::algorithm::contains(mol->getStereoGroups()[0].getAtoms(),
+                                        std::vector<Atom *>{atom})) {
+          isSimple = false;
+          break;
+        }
+      }
+      if (isSimple) {
+        for (auto &bond : mol->bonds()) {
+          if ((bond->getStereo() == Bond::BondStereo::STEREOATROPCW ||
+               bond->getStereo() == Bond::BondStereo::STEREOATROPCCW) &&
+              !boost::algorithm::contains(mol->getStereoGroups()[0].getBonds(),
+                                          std::vector<Bond *>{bond})) {
+            isSimple = false;
+            break;
+          }
+        }
+      }
```

Comment:
I like the use of find when there is only one item to check for containment rather than BOOST. I changed these as suggested.  In addition, there were several other uses of BOOST to determine containment, and I changed them to use std::find and std::find_if.  These occurred in  MarvinDefs.cpp